doc/src/modules/polys/numberfields.rst

```diff
@@ -0,0 +1,289 @@
+
+
+As points in the complex plane
+``````````````````````````````
+
+The minimal polynomial for an algebraic number determines it only up to
+conjugacy; in other words, the number and all its conjugates are algebraically
+indistinguishable.
+
+In order to select a particular conjugate, we can identify
+it with a point in the complex plane. This means identifying two real
+intervals, one on the real axis and one on the imaginary axis, such that
+exactly one root of the given irreducible polynomial over $\mathbb{Q}$ lies
+within the complex region so defined.
+
+Support for this in SymPy is currently limited. The :py:func:`~.isolate`
+function is currently applicable only to algebraic numbers lying on the
+real line.
+
+.. autofunction:: isolate
+
+
+As elements of finitely-generated modules
+`````````````````````````````````````````
+
+In computational algebraic number theory, finitely-generated
+$\mathbb{Z}$-modules are of central importance. For example, every
+order_ and every ideal_ is such a module.
+
+In particular, the maximal order -- or `ring of integers`_ -- in a number field
+is a finitely-generated $\mathbb{Z}$-module, whose generators form an
+`integral basis`_ for the field.
+
+Classes allowing us to represent such modules, and their elements, are provided
+in the :py:mod:`~.modules` module. Here, the :py:class:`~.ModuleElement` class
+provides another way to represent algebraic numbers.
+
+
+Finitely-generated modules
+--------------------------
+
+.. automodule:: sympy.polys.numberfields.modules
+
+Class Reference
+```````````````
+.. autoclass:: Module
+   :members:
+
+   .. automethod:: Module.__call__
+
+.. autoclass:: PowerBasis
+   :members:
+
+   .. automethod:: PowerBasis.__init__
+
+.. autoclass:: Submodule
+   :members:
+
+   .. automethod:: Submodule.__init__
+
+.. autoclass:: ModuleElement
+   :members:
+
+   .. automethod:: ModuleElement.__init__
+   .. automethod:: ModuleElement.__add__
+   .. automethod:: ModuleElement.__mul__
+   .. automethod:: ModuleElement.__mod__
+
+.. autoclass:: PowerBasisElement
+   :members:
+
+.. autofunction:: make_mod_elt
+
+.. autoclass:: ModuleHomomorphism
+   :members:
+
+   .. automethod:: ModuleHomomorphism.__init__
+
+.. autoclass:: ModuleEndomorphism
+   :members:
+
+   .. automethod:: ModuleEndomorphism.__init__
+
+.. autoclass:: InnerEndomorphism
+   :members:
+
+   .. automethod:: InnerEndomorphism.__init__
+
+.. autoclass:: EndomorphismRing
+   :members:
+
+   .. automethod:: EndomorphismRing.__init__
+
+.. autofunction:: find_min_poly
+
+
+Utilities
+---------
+
+.. currentmodule:: sympy.polys.numberfields.utilities
+
+.. autofunction:: is_rat
+.. autofunction:: is_int
+.. autofunction:: get_num_denom
+.. autofunction:: extract_fundamental_discriminant
+
+.. autoclass:: AlgIntPowers
+   :members:
+
+   .. automethod:: AlgIntPowers.__init__
+
+.. autofunction:: coeff_search
+.. autofunction:: supplement_a_subspace
+
+
+Solving the Main Problems
+=========================
```

Comment:
Reading this in the rendered docs this title seems odd coming after the reference docs for the classes and functions above. I wonder if this belongs further up.